### ISSUE TITLE
RCS1248 triggers when used in expression trees, but pattern-matching 'is' operator cannot be used in expression trees

### ISSUE THREAD

**Issue by Jaap-van-Hengstum** [JosefPihrt/Roslynator#901]
**Product and Version Used**: 
4.1.0

**Steps to Reproduce**:
```
roslynator_null_check_style = pattern_matching
dotnet_diagnostic.RCS1248.severity = suggestion # "Normalize null check"
```

**Actual Behavior**:
```csharp
var users = dbContext.Users
    .Where(user => user.Address != null) // <-- RCS1248 triggers here
    .ToList();
```
(using [Queryable.Where](https://docs.microsoft.com/en-us/dotnet/api/system.linq.queryable.where?view=net-6.0))

**Expected Behavior**:
RCS1248 doesn't trigger when used in expression tree.
